src/monodroid/jni/cpp-util.hh

```diff
@@ -1,7 +1,41 @@
 #ifndef __CPP_UTIL_HH
 #define __CPP_UTIL_HH
 
-#include <cppcompat.hh>
+#include <cstdarg>
+#include <cstdlib>
+
+#if defined (ANDROID)
+#include <android/log.h>
+#else
+#include <cstdio>
+#endif
+
+#include "cppcompat.hh"
+#include "platform-compat.hh"
+
+static inline void
+do_abort_unless (bool condition, const char* fmt, ...)
+{
+	if (XA_LIKELY (condition)) {
+		return;
+	}
+
+	va_list ap;
+
+	va_start (ap, fmt);
+#if defined (ANDROID)
+	__android_log_vprint (ANDROID_LOG_FATAL, "monodroid", fmt, ap);
+#else
+	vfprintf (stderr, fmt, ap);
```

Comment:
Shouldn't this also call `fprintf (stderr, "\n")`?  Very often these messages will *not* have a trailing `\n`, and stderr is *line buffered* by default (IIRC), so for sanity don't we want to emit a final `\n`?